### PR TITLE
add ignore.fixed.terms and fix.intercept.variance params.

### PR DESCRIPTION
**Pull request by jamesonquinn**
bug https://github.com/jknowles/merTools/issues/75

### CODE REVIEW

**Comment by jknowles** [comment]
@jamesonquinn This looks pretty good.

Can you add a couple of things:

- A citation (if possible) that talks about this method of adjusting the intervals, or even a link to an online write-up 
- A unit test that shows different output when the user sets this option and when they do not; the example you use here I will also probably include at the end of a vignette and in the `README` to show off this new feature

**Comment by jamesonquinn** [comment]
Done, I think. How does it look?

(I have no citation; I worked out this correction by myself.)

**Comment by jknowles** [comment]
Thanks @jamesonquinn  I will wait for @carlbfrederick to weigh in to make sure it doesn't break any future plans we might have had and that the correction makes sense to him too

**Comment by carlbfrederick** [comment]
Sorry for the delay @jamesonquinn and @jknowles. This is on my to do list for tomorrow :)

File: R/merPredict.R [hunk 1]
```diff
@@ -83,7 +93,10 @@ predictInterval <- function(merMod, newdata, which=c("full", "fixed", "random",
                             n.sims = 1000, stat=c("median","mean"),
                             type=c("linear.prediction", "probability"),
                             include.resid.var=TRUE, returnSims = FALSE,
-                            seed=NULL, .parallel = FALSE, .paropts = NULL){
+                            seed=NULL, .parallel = FALSE, .paropts = NULL,
+                            fix.intercept.variance = FALSE, #This does NOT work with random slope models
```
Comment (carlbfrederick) [inline R/merPredict.R]: I would make this limitation explicit in the roxygen description of the fix.intercept.variance parameter above.

File: R/merPredict.R [hunk 2]
```diff
@@ -254,6 +267,41 @@ predictInterval <- function(merMod, newdata, which=c("full", "fixed", "random",
...
   vcov.tmp <- as.matrix(vcov(merMod))
+  if (fix.intercept.variance) {
+    #Assuming all random effects include intercepts.
+    intercept.variance = vcov.tmp[1,1]
+
+    groupsizes = ngrps(merMod)
+    for(j in names(groupsizes)){ #for every group of random e
+
+      groupExtraPrecision = 0
+      groupVar = (attr(VarCorr(cvoteg)[[j]],"stddev")["(Intercept)"])^2
```
Comment (carlbfrederick) [inline R/merPredict.R]: Appears that `cvoteg` should be `merMod`, no?  Is this the name of the model you tested on?

File: R/merPredict.R [hunk 3]
```diff
@@ -254,6 +267,41 @@ predictInterval <- function(merMod, newdata, which=c("full", "fixed", "random",
   ##Calculate yhat as sum of the components (fixed plus all groupling factors)
   fe.tmp <- fixef(merMod)
   vcov.tmp <- as.matrix(vcov(merMod))
+  if (fix.intercept.variance) {
+    #Assuming all random effects include intercepts.
+    intercept.variance = vcov.tmp[1,1]
+
+    groupsizes = ngrps(merMod)
+    for(j in names(groupsizes)){ #for every group of random e
```
Comment (carlbfrederick) [inline R/merPredict.R]: Is it problematic that we are iterating over `j` in nested loops (lines 152 and 275) or is this on purpose?

File: R/merPredict.R (hunk 1 above)
Comment (carlbfrederick) [inline R/merPredict.R]: re: #77 I would make this limitation explicit in the roxygen description of the fix.intercept.variance parameter above. @jknowles @jamesonquinn

Comment (jknowles) [inline R/merPredict.R]: I agree with this, let us know when you can do this @jamesonquinn 

Then add yourself as a contributor to the DESCRIPTION

**Comment by jknowles** [comment]
@jamesonquinn can you make the changes @carlbfrederick pointed out and push again?

**Comment by jknowles** [comment]
Sorry @jamesonquinn -- should have pinged you here. 🤕

File: vignettes/Using_predictInterval.Rmd [hunk 4]
```diff
@@ -201,9 +201,9 @@ effects as fully-known (that is, with an effectively 0 variance.) This is done
 using the `ignore.fixed.effects` argument.
 
 ```{r ShowFat2, message=FALSE, warning=FALSE}
-predictInterval(ifmodel, ais[1,], include.resid.var=0, ignore.fixed.terms = 1) #predict the average body fat for a group of 196cm female baseball players, taking the global intercept (mean body fat) as fully known
-predictInterval(ifmodel, ais[1,], include.resid.var=0, ignore.fixed.terms = "(Intercept)") #Same as above
-predictInterval(ifmodel, ais[1,], include.resid.var=0, ignore.fixed.terms = 1:2) #as above, taking the first two fixed effects (intercept and height effect) as fully known
+predictInterval(fatmodel, ais[1,], include.resid.var=0, ignore.fixed.terms = 1) #predict the average body fat for a group of 196cm female baseball players, taking the global intercept (mean body fat) as fully known
```
Comment (jknowles) [inline vignettes/Using_predictInterval.Rmd]: Hi @jamesonquinn  -- where does the `ais` dataset come from? Need to include it in the full unit test suite and I can't seem to find it.

**Pull request merged**